### PR TITLE
Align Python/C compiler API with command line usage.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Normalizes flag/option handling for both to share the command line flag syntax, removing the partial existing support for individual options in the C/Python API.
* Rename iree.compiler.api.driver -> iree.compiler.transforms.ireec to better layer with upstream and represent that this is the programmatic analog to the ireec CL tool.
* Also adds CMake option to allow disabling `iree.jax` legacy API as it is getting in the way of its out of tree replacement and didn't belong in tree like this anyway.

### CODE REVIEW

File: iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertHALToVM.cpp [hunk 1]
```diff
@@ -138,7 +138,7 @@ std::unique_ptr<OperationPass<ModuleOp>> createConvertHALToVMPass(
 }
 
 static PassRegistration<ConvertHALToVMPass> pass([] {
-  auto options = IREE::VM::getTargetOptionsFromFlags();
+  auto options = IREE::VM::TargetOptions::FromFlags::getRegistered();
```
Comment (benvanik) [inline iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertHALToVM.cpp:141]: `getRegistered()` reads really weird to me as I'd think that meant "get the list of registered options" vs "get the options that are registered" - maybe just rename to `get()`?

**Review by benvanik — APPROVED** [review]

**Comment by benvanik** [comment]
(also is that pytype error legit?)

**Comment by stellaraccident** [comment]
> (also is that pytype error legit?)

No. pytype gets cranky easily and this file should just be excluded entirely.

File: iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertHALToVM.cpp (hunk 1 above)
Comment (stellaraccident) [reply iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertHALToVM.cpp:141]: Thanks - I'll make that change.

**Pull request merged**